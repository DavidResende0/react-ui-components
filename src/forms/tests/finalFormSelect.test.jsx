import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { FinalFormSelect } from '../';
import { fieldMetaProps, fieldInputProp } from './mocks';

describe('React select component', () => {
  let initialProps = {};
  let options = [];
  beforeEach(() => {
    initialProps = { meta: fieldMetaProps, input: fieldInputProp, validateOnMount: true };
    options = [{
      label: 'option 1',
      value: 1,
    }, {
      label: 'option 2',
      value: 2,
    }];
  });

  it('Should render correctly', () => {
    const tree = shallow(<FinalFormSelect
      label="test select"
      options={options}
      {...initialProps}
    />).dive();
    expect(toJson(tree)).toMatchSnapshot();
  });

  it('Should render correctly in error state', () => {
    const meta = { ...fieldMetaProps };
    meta.error = 'Required';
    initialProps = { ...initialProps, meta };
    const tree = shallow(<FinalFormSelect
      label="test select"
      options={options}
      {...initialProps}
    />).dive();
    expect(toJson(tree)).toMatchSnapshot();
  });
});
