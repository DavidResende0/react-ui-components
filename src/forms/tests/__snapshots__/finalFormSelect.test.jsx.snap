// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`React select component Should render correctly 1`] = `
<FormGroup
  bsClass="form-group"
  validationState={null}
>
  <Col
    bsClass="col"
    className="control-label"
    componentClass="label"
    xs={2}
  >
    test select
  </Col>
  <Col
    bsClass="col"
    componentClass="div"
    xs={8}
  >
    <Select
      arrowRenderer={[Function]}
      autosize={true}
      backspaceRemoves={true}
      backspaceToRemoveMessage="Press backspace to remove {label}"
      className=" final-form-select"
      clearAllText="Clear all"
      clearRenderer={[Function]}
      clearValueText="Clear value"
      clearable={false}
      closeOnSelect={true}
      deleteRemoves={true}
      delimiter=","
      disabled={false}
      escapeClearsValue={true}
      filterOptions={[Function]}
      id="name"
      ignoreAccents={true}
      ignoreCase={true}
      inputProps={Object {}}
      isLoading={false}
      joinValues={false}
      label="test select"
      labelKey="label"
      matchPos="any"
      matchProp="any"
      menuBuffer={0}
      menuRenderer={[Function]}
      multi={false}
      name="name"
      noResultsText="No results found"
      onBlur={[MockFunction]}
      onBlurResetsInput={true}
      onChange={[Function]}
      onCloseResetsInput={true}
      onFocus={[MockFunction]}
      onSelectResetsInput={true}
      openOnClick={true}
      optionClassName="final-form-select-option"
      optionComponent={[Function]}
      options={
        Array [
          Object {
            "label": "option 1",
            "value": 1,
          },
          Object {
            "label": "option 2",
            "value": 2,
          },
        ]
      }
      pageSize={5}
      placeholder=""
      removeSelected={true}
      required={false}
      rtl={false}
      scrollMenuIntoView={true}
      searchable={false}
      simpleValue={false}
      tabSelectsValue={true}
      trimFilter={true}
      value=""
      valueComponent={[Function]}
      valueKey="value"
    />
  </Col>
</FormGroup>
`;

exports[`React select component Should render correctly in error state 1`] = `
<FormGroup
  bsClass="form-group"
  validationState="error"
>
  <Col
    bsClass="col"
    className="control-label"
    componentClass="label"
    xs={2}
  >
    test select
  </Col>
  <Col
    bsClass="col"
    componentClass="div"
    xs={8}
  >
    <Select
      arrowRenderer={[Function]}
      autosize={true}
      backspaceRemoves={true}
      backspaceToRemoveMessage="Press backspace to remove {label}"
      className=" final-form-select"
      clearAllText="Clear all"
      clearRenderer={[Function]}
      clearValueText="Clear value"
      clearable={false}
      closeOnSelect={true}
      deleteRemoves={true}
      delimiter=","
      disabled={false}
      escapeClearsValue={true}
      filterOptions={[Function]}
      id="name"
      ignoreAccents={true}
      ignoreCase={true}
      inputProps={Object {}}
      isLoading={false}
      joinValues={false}
      label="test select"
      labelKey="label"
      matchPos="any"
      matchProp="any"
      menuBuffer={0}
      menuRenderer={[Function]}
      multi={false}
      name="name"
      noResultsText="No results found"
      onBlur={[MockFunction]}
      onBlurResetsInput={true}
      onChange={[Function]}
      onCloseResetsInput={true}
      onFocus={[MockFunction]}
      onSelectResetsInput={true}
      openOnClick={true}
      optionClassName="final-form-select-option"
      optionComponent={[Function]}
      options={
        Array [
          Object {
            "label": "option 1",
            "value": 1,
          },
          Object {
            "label": "option 2",
            "value": 2,
          },
        ]
      }
      pageSize={5}
      placeholder=""
      removeSelected={true}
      required={false}
      rtl={false}
      scrollMenuIntoView={true}
      searchable={false}
      simpleValue={false}
      tabSelectsValue={true}
      trimFilter={true}
      value=""
      valueComponent={[Function]}
      valueKey="value"
    />
    <HelpBlock
      bsClass="help-block"
    >
      Required
    </HelpBlock>
  </Col>
</FormGroup>
`;
